src/main/java/io/supertokens/inmemorydb/queries/PasswordlessQueries.java

```diff
@@ -149,23 +151,48 @@ public static void deleteDevice_Transaction(Start start, Connection con, String
             throws SQLException {
-        String QUERY = "DELETE FROM " + Config.getConfig(start).getPasswordlessDevicesTable()
+        // We lock the whole table instead of the individual devices. This is not intended for production use.
+        ((ConnectionWithLocks) con).lock(Config.getConfig(start).getPasswordlessDevicesTable());
+
+        // SQLite is not compiled with foreign key constraint and so we must implement cascading deletes here
+        String QUERY = "SELECT device_id_hash FROM " + Config.getConfig(start).getPasswordlessDevicesTable()
                 + " WHERE phone_number = ?";
 
+        List<String> deviceIdHashes = new ArrayList<>();
         try (PreparedStatement pst = con.prepareStatement(QUERY)) {
             pst.setString(1, phoneNumber);
-            pst.executeUpdate();
+            ResultSet result = pst.executeQuery();
+            while (result.next()) {
+                deviceIdHashes.add(result.getString("device_id_hash"));
+            }
         }
+        deleteRowsByDeviceIdHashList_Transaction(con, Config.getConfig(start).getPasswordlessCodesTable(),
+                deviceIdHashes);
+        deleteRowsByDeviceIdHashList_Transaction(con, Config.getConfig(start).getPasswordlessDevicesTable(),
+                deviceIdHashes);
     }
 
     public static void deleteDevicesByEmail_Transaction(Start start, Connection con, @Nonnull String email)
-            throws SQLException {
-        String QUERY = "DELETE FROM " + Config.getConfig(start).getPasswordlessDevicesTable() + " WHERE email = ?";
+            throws SQLException, StorageQueryException {
+        // We lock the whole table instead of the individual devices. This is not intended for production use.
+        ((ConnectionWithLocks) con).lock(Config.getConfig(start).getPasswordlessDevicesTable());
+
+        // SQLite is not compiled with foreign key constraint and so we must implement cascading deletes here
+        String QUERY = "SELECT device_id_hash FROM " + Config.getConfig(start).getPasswordlessDevicesTable()
+                + " WHERE email = ?";
 
+        List<String> deviceIdHashes = new ArrayList<>();
         try (PreparedStatement pst = con.prepareStatement(QUERY)) {
             pst.setString(1, email);
-            pst.executeUpdate();
+            ResultSet result = pst.executeQuery();
+            while (result.next()) {
+                deviceIdHashes.add(result.getString("device_id_hash"));
+            }
         }
+        deleteRowsByDeviceIdHashList_Transaction(con, Config.getConfig(start).getPasswordlessCodesTable(),
+                deviceIdHashes);
+        deleteRowsByDeviceIdHashList_Transaction(con, Config.getConfig(start).getPasswordlessDevicesTable(),
+                deviceIdHashes);
     }
 
     public static void createCode_Transaction(Start start, Connection con, PasswordlessCode code) throws SQLException {
```

Comment:
If this function is not being used anywhere outside this file, either:
- make it a part of the function which is calling it (in case it's just one function)
- Make this `private static void`